src/v/cluster/controller_backend.cc

```diff
@@ -863,6 +878,27 @@ ss::future<> controller_backend::remove_from_shard_table(
       });
 }
 
+ss::future<std::error_code> controller_backend::create_non_replicable_partition(
+  model::ntp ntp, model::revision_id rev) {
+    auto cfg = _topics.local().get_topic_cfg(model::topic_namespace_view(ntp));
+    if (!cfg) {
+        // partition was already removed, do nothing
+        co_return errc::success;
+    }
+    if (likely(!_storage.local().log_mgr().get(ntp))) {
+        auto ntp_cfg = cfg->make_ntp_config(
+          _data_directory, ntp.tp.partition, rev);
+        co_await _storage.local().log_mgr().manage(std::move(ntp_cfg));
+    } else {
+        auto sr = _shard_table.local().shard_revision_for(ntp);
+        if (sr && sr->revision < rev) {
+            co_return errc::partition_already_exists;
```

Comment:
I think that here we shouldn't return `errc::partition_already_exists`. This error is not  going to be solved in any further reconciliation loop iterations. The loop will stuck in this state i.e. `check if exists-> exists->error`. Maybe it would be better to assert in here since this is logical error.